File: frontend/src/services/akgda/models/Project.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Layer } from './Layer';

export type Project = {
    readonly id: number;
    readonly layers: Array<Layer>;
    pid: string;
    name: string;
    description: string;
    project_type: string;
};

File: backend/api/app/schemas/project.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict

from app.schemas.layer import LayerSchema


class ProjectBase(BaseModel):
    pid: str
    name: str
    description: str | None = None
    project_type: str


class ProjectCreate(ProjectBase):
    layer_ids: list[int] = []


class ProjectUpdate(BaseModel):
    pid: str | None = None
    name: str | None = None
    description: str | None = None
    project_type: str | None = None
    layer_ids: list[int] | None = None


class ProjectSchema(ProjectBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    layers: list[LayerSchema] = []
    created_at: datetime
    updated_at: datetime | None = None
